packages/aws-cdk/lib/api/evaluate-cloudformation-template.ts

```diff
@@ -1,6 +1,38 @@
 import * as cxapi from '@aws-cdk/cx-api';
 import * as AWS from 'aws-sdk';
-import { ListStackResources } from './common';
+import { ISDK } from './aws-auth';
+
+export interface ListStackResources {
+  listStackResources(): Promise<AWS.CloudFormation.StackResourceSummary[]>;
+}
+
+export class LazyListStackResources implements ListStackResources {
```

Comment:
> Nit: it's more like `CachedListStackResources` than `Lazy*`.

It is actually Lazy though (it will not make a service call unless you call `listResources()`).

Reply:
That is true for most methods though 🤣

Reply:
Not sure what you mean. I think a very sensible non-lazy implementation would be to make the service call in the constructor of the class, and not in `listResources()` (it would be simpler, for one thing).